_spinnaker/pacrd.md

```diff
@@ -0,0 +1,372 @@
+```
+
+### Updating
+
+Applications can be updated in one of two ways:
+
+1. Re-applying the application manifest in your repository
+    - `kubectl apply -f application.yaml`
+1. Editing the application manifest in-cluster
+    - `kubectl edit app myapplicationname`
+
+In both cases, when an application has been updated in your Kubernetes cluster,
+you will see changes propagate into Spinnaker shortly. If an error occurred
+during the update your application may show an `ErrorFailedUpdate` state. You
+can see the details of that failure by describing the resource and looking in
+the "Events" section:
+
+```
+kubectl describe app myapplicationname
+```
+
+### Deleting
+
+When an application is no longer necessary, you can delete it in one of two ways:
+
+1. Re-applying the application manifest in your repository
+  - `kubectl delete -f application.yaml`
+1. Deleting the application directly
+  - `kubectl delete app myapplicationname`
+
+In either case the application will be deleted shortly in Spinnaker as well. If
+an error occurred during deletion then your application may show an
+`ErrorFailedDelete` state. You can see the details of that failure by
+describing the resource and looking in the "Events section":
+
+```
+kubectl describe app myapplicationname
+```
+
+#### Caveats
+
+For users of version `v0.1.x` deleting an application in Kubernetes will trigger
+the following behavior:
+
+- Delete the application in Kubernetes.
+- Delete the application in Spinnaker.
+- Delete pipelines associated with the application _in Spinnaker only_.
+
+This is a known gap in the behavior of application deletion and will be fixed
+in a future release.
+
+## Pipelines
+
+Pipelines allow you to encode the process that your team follows to take a
+service from commit to a desired environment (such as production). You can
+read more about [pipelines in the upstream Spinnaker documentation][pipe-docs].
+
+### Current Limitations
+
+The following sections document limitations with various versions of the PaCRD
+controller for this resource.
+
+#### v0.1.x
+
+- Pipeline stages must be defined with a `type` key for the stage name and a
+key of the same name where all stage options live. For example, for the
+"Bake Manifest" stage you would structure your definition like so:
+  
+```yaml
+# ...
+stages:
+  - type: BakeManifest
+    bakeManifest:
+      name: Bake the Bread
+      # ...
+# ...
+```
+
+- Documentation for available Pipeline spec fields must currently be
+found in the installation manifest for this controller. You can do so by
+grepping for `pipelines.pacrd.armory.spinnaker.io` in the installation
+manifest. Fields are documented under `spec.validation.openAPIV3Schema`.
+
+### Creating
+
+Pipelines in Kubernetes can be defined with the same options you would define
+in the UI. The following example shows how you might define a trivial Pipeline:
```

Comment:
I'd describe what the pipeline does instead of saying `trivial`, so

"The following example shows how you might define a simple pipeline that bakes a manifest and prompts for a manual judgement: